### PR TITLE
Fix for issue 86

### PR DESCRIPTION
**Pull request by fraenky8**
Refactored `changeMapToURLValues()` and added tests.

### CODE REVIEW

**Comment by fraenky8** [comment]
Why have the checks failed? On my computer i ran `go test -v -run ChangeMapToURLValues` and all tests pass. Using `go version go1.6 windows/amd64`

**Comment by fraenky8** [comment]
Last commit should not influenced the travis-build, therefore i assume the test fails from time to time (here local too, sometimes it works, sometimes not).

File: main.go [hunk 1]
```diff
@@ -591,16 +591,65 @@ func changeMapToURLValues(data map[string]interface{}) url.Values {
...
-			}
+		case bool:
+			newUrlValues.Add(k, strconv.FormatBool(val))
 		// if a number, change to string
 		// json.Number used to protect against a wrong (for GoRequest) default conversion
 		// which always converts number to float64.
 		// This type is caused by using Decoder.UseNumber()
 		case json.Number:
 			newUrlValues.Add(k, string(val))
+		case int:
```
Comment (QuentinPerez) [inline main.go]: case int, int32, int64:

File: main.go [hunk 2]
```diff
@@ -591,16 +591,65 @@ func changeMapToURLValues(data map[string]interface{}) url.Values {
...
+			newUrlValues.Add(k, strconv.FormatInt(int64(val), 10))
+		// TODO add all other int-Types (int8, int16, ...)
+		case float64:
```
Comment (QuentinPerez) [inline main.go]: case float64, float32:

File: main.go [hunk 3]
```diff
@@ -591,16 +591,65 @@ func changeMapToURLValues(data map[string]interface{}) url.Values {
...
+			newUrlValues.Add(k, strconv.FormatFloat(float64(val), 'f', -1, 64))
+		case float32:
+			newUrlValues.Add(k, strconv.FormatFloat(float64(val), 'f', -1, 64))
+		// following slices are mostly needed for tests
+		case []string:
+			for _, element := range val {
+				newUrlValues.Add(k, element)
+			}
+		case []int:
```
Comment (QuentinPerez) [inline main.go]: case []int, []int32, []int64:

File: main.go [hunk 4]
```diff
@@ -591,16 +591,65 @@ func changeMapToURLValues(data map[string]interface{}) url.Values {
...
+			for _, element := range val {
+				newUrlValues.Add(k, strconv.FormatInt(int64(element), 10))
+			}
+		case []bool:
+			for _, element := range val {
+				newUrlValues.Add(k, strconv.FormatBool(element))
+			}
+		case []float64:
```
Comment (QuentinPerez) [inline main.go]: case []float64, []float32:

**Comment by QuentinPerez** [comment]
LGTM thank you 😄

**Pull request merged**